fms/models/llama.py

```diff
@@ -233,12 +237,17 @@ def get_config(self) -> LLaMAConfig:
     def from_config(cls, config: LLaMAConfig) -> "LLaMA":
         return cls(config)
 
-    def reset_params(self):
-        # Modules are self-initializing, we're just going to down-scale the final prediction head to be
-        # mixed-fan (inputs and gradients scale to the same inverse factors) if it isn't tied
-        self.shared.head.weight.data.normal_(
-            0, 1 / math.sqrt(math.sqrt(self.width * self.shared.vocab_size))
-        )
+    def reset_params(self, orig_init):
+        if not orig_init:
+            # Modules are self-initializing, we're just going to down-scale the final prediction head to be
+            # mixed-fan (inputs and gradients scale to the same inverse factors)
+            self.shared.head.weight.data.normal_(
+                0, 1 / math.sqrt(math.sqrt(self.width * self.shared.vocab_size))
+            )
+        else:
```

Comment:
I don't know of us depending on the above behavior. should we just make the below be th default instead of adding a new parameter?

Reply:
If we want to support depth-correction, we wouldn't do it with the below init, so we would still need a flag. If we solely care about reproducing Llama exactly, though, I could strip out the depth-correction and main init schemes and leave only the below hard-coded

Reply:
Let's only support what worked well, no need to add other options.

Reply:
I talked to Linsong - we'll minimize changes and remove `orig_init` entirely, and add it back later as an entirely separate `post_process_init`-ish function in the new repo (since we used it only for the pretraining portions anyways)

Reply:
I think I'd also favor just hard-coding whatever we know works well, unless we know we have a use-case for multiple init schemes